manifests/init.pp

```diff
@@ -460,17 +462,32 @@
       }
     }
 
-    python::pip { 'patroni':
-      ensure      => $version,
-      virtualenv  => $install_dir,
-      environment => ["PIP_PREFIX=${install_dir}"],
-      before      => File['patroni_config'],
-    }
+    if $custom_pip_provider != undef {
```

Comment:
Could simplify this a little and just say `if $custom_pip_provider` , Undef is sort of like nil/false so it will behave the same. I have some vague recollection that testing against undef actually has to be more against the data type like this: `if $custom_pip_provider !~ Undef`, but that's if you want to test against hte data type.